boot/bootutil/src/loader.c

```diff
@@ -756,8 +763,15 @@ boot_validate_slot(struct boot_loader_state *state, int slot,
         }
     }
 #endif
-
-    FIH_CALL(boot_image_check, fih_rc, state, hdr, fap, bs);
+#ifdef MCUBOOT_IMAGE_ACCESS_HOOKS
+    if (slot == 0) {
```

Comment:
instead of having this if slot check could we just pass the slot to the function like this
`boot_image_check_hook(int slot, int img_index);`

Reply:
sure, will change this as you are suggesting.

Reply:
Also remove `if (slot == 0) {` now.

Reply:
Done